examples/dynamic_dlopen/impl.h

```diff
@@ -0,0 +1,90 @@
+#ifndef EXAMPLES_DYNAMIC_DLOPEN_IMPL_H
+#define EXAMPLES_DYNAMIC_DLOPEN_IMPL_H
+
+#include "../../blocks/http/api.h"
+#include "../../bricks/file/file.h"
+#include "../../bricks/system/syscalls.h"
+#include "../../bricks/util/sha256.h"
+#include "../../typesystem/serialization/json.h"
+#include "../../typesystem/struct.h"
+
+#include "../iris/data/dataset.h"
+
+class DynamicDLOpenIrisExampleImpl {
+ private:
+  using F = void (*)(const Schema&, std::ostringstream&);
+
+  // clang-format off
+  const std::string prefix_ = current::Base64Decode("I2luY2x1ZGUgPHZlY3Rvcj4KI2luY2x1ZGUgPHNzdHJlYW0+CiNpbmNsdWRlIDxpb21hbmlwPgoKc3RydWN0IEJhc2UgewogIHZpcnR1YWwgfkJhc2UoKSB7fQp9OwoKc3RydWN0IEZsb3dlciA6IEJhc2UgewogIHZpcnR1YWwgfkZsb3dlcigpIHt9CiAgdW5pb24gewogICAgc3RydWN0IHsKICAgICAgZG91YmxlIHNsOwogICAgICBkb3VibGUgc3c7CiAgICAgIGRvdWJsZSBwbDsKICAgICAgZG91YmxlIHB3OwogICAgfTsKICAgIGRvdWJsZSB4WzRdOwogIH07CiAgc3RkOjpzdHJpbmcgbGFiZWw7Cn07CgpleHRlcm4gIkMiIHZvaWQgUnVuKGNvbnN0IHN0ZDo6dmVjdG9yPEZsb3dlcj4mIGZsb3dlcnMsIHN0ZDo6b3N0cmluZ3N0cmVhbSYgb3MpIHsK");
+  const std::string suffix_ = "\n}\n";
```

Comment:
Use `{prefix,suffix}.inl` instead of hard coding huge base64 strings?

Reply:
Then the code would become base-directory-dependent, or I'd have to incorporate the `argv[0]` logic, not to mention the binary would stop working by itself.

I could add a unit test to confirm these constants are exactly the base64 representations of the `.inl` files, what do you think about it?

Reply:
@mzhurovich, a better idea could be to augment our default `Makefile[Impl]` with the logic to wrap up every `.inl` within the directory into some `.gitignore`-d header file containing their base64 representations prior to any compilation.

This way we get the best of all three worlds: The "constants" are automatically initialized with the up-to-date `.inl`-files, the `.inl` files can even be binary, and yet the resulting output binaries are self-contained, with no dependencies on any external files.

That's how it's done for static content in FD, and I've had no issues whatsoever, on both Linux and Mac.

Again, now doesn't seem the right time to be doing that. Just an option to keep in mind.